CMakeLists.txt

```diff
@@ -570,6 +570,12 @@ add_custom_command(TARGET libopenmc POST_BUILD
   ${CMAKE_CURRENT_SOURCE_DIR}/openmc/lib/$<TARGET_FILE_NAME:libopenmc>
   COMMENT "Copying libopenmc to Python module directory")
 
+add_custom_command(TARGET openmc POST_BUILD
+  COMMAND ${CMAKE_COMMAND} -E copy
+  $<TARGET_FILE:openmc>
+  ${CMAKE_CURRENT_SOURCE_DIR}/openmc/lib/$<TARGET_FILE_NAME:openmc>
+  COMMENT "Copying openmc binary to Python module directory")
+
```

Comment:
I know that the `libopenmc` library already gets relocated into the source tree, but IMHO that's always felt a bit "unclean" to put build artifacts into the source tree. 

This might be a chance to move away from that approach. 
If you are interested, [in my attempt at creating a pip package](https://github.com/hassec/openmc/blob/hassec_setuptools/CMakeLists.txt) I had made changes to the [`setup.py`](https://github.com/hassec/openmc/blob/hassec_setuptools/setup.py) to include the build step by implementing a `setuptools.Extension` for OpenMC which builds in a temporary directory.
 
That's one way, I'm sure there is more 🤷

Reply:
Yes, what we have currently for libopenmc is very hacky, so I would not advocate for doubling down on it and handling the executable the same way. The approach you have there @hassec sounds a lot better. The other option that comes to mind is to use scikit-build (see #916). It looks like there are some [other build backends](https://github.com/scikit-build/scikit-build-core#other-projects-for-building) out there that might merit a look.

Reply:
FWIW, @shimwell had a version that tried to use `scikit-build`, and I started out with that approach in the beginning but couldn't really get it to work as well as with the `setuptools.Extension` which was more flexible and easier to setup as `scikit-build` had limited docs. But there as been a lot of recent activity on the new [scikit-build-core](https://github.com/scikit-build/scikit-build-core). 

I don't have any experience with that package yet.